### PR TITLE
fix removing double slash prefix in raw req path

### PR DESCRIPTION
**Pull request by RamanaReddy0M**
## Proposed changes

#3924 Raw Request removes // (double slash) in Request 

<details>
  <summary>Test template</summary>
  
  ```yaml
  id: remove-path

info:
  name: Path // Remove in Raw Request
  author: DhiyaneshDK
  severity: info

http:
  - raw:
      - |
        POST //CFIDE/wizards/common/utils.cfc?method=wizardHash^&inPassword=foo^&_cfclient=true^&returnFormat=wddx HTTP/1.1
        Host: {{Hostname}}
        Content-Type: application/x-www-form-urlencoded

        _variables={"about":{"_metadata":{"classname":"\..\lib\password.properties"},"_variables":{}}}

    matchers-condition: and
    matchers:
      - type: word
        part: body
        words:
          - '<wddcPacket'
          - '<struct>'
        condition: and

      - type: status
        status:
          - 200
 
  ```

</details>

## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [ ] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by RamanaReddy0M** [comment]
```console
✗ ./nuclei -u https://example.com -t test.yaml -debug-req

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.9.9

                projectdiscovery.io

[INF] Current nuclei version: v2.9.9 (latest)
[INF] Current nuclei-templates version: v9.5.8 (latest)
[INF] New templates added in latest release: 113
[INF] Templates loaded for current scan: 1
[INF] Targets loaded for current scan: 1
[INF] [remove-path] Dumped HTTP request for https://example.com//CFIDE/wizards/common/utils.cfc?method=wizardHash^&inPassword=foo^&_cfclient=true^&returnFormat=wddx

POST //CFIDE/wizards/common/utils.cfc?method=wizardHash^&inPassword=foo^&_cfclient=true^&returnFormat=wddx HTTP/1.1
Host: example.com
User-Agent: Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/35.0.3319.102 Safari/537.36
Connection: close
Content-Length: 94
Content-Type: application/x-www-form-urlencoded
Accept-Encoding: gzip

_variables={"about":{"_metadata":{"classname":"\..\lib\password.properties"},"_variables":{}}}
[INF] No results found. Better luck next time!
```

**Review by tarunKoyalwar — CHANGES_REQUESTED** [review]
i was trying to reproduce this issue when it was created . i found that the issue was actually in `cloned.MergePath` method which takes care of merging two paths .  while whitelisting edgecases will resolve this in nuclei but i think it will be good if we handle such cases in urlutil itself . what do you think @Mzack9999 @RamanaReddy0M 

Not sure but since this issue is in lib itself . non-raw/ base requests might also have this issue

i think the issue is somewhere in this function https://github.com/projectdiscovery/utils/blob/5efadaf067b4ba4da7149465e52361449e46cce1/url/utils.go#L38-L72 .

**Comment by RamanaReddy0M** [comment]
> i was trying to reproduce this issue when it was created . i found that the issue was actually in `cloned.MergePath` method which takes care of merging two paths . while whitelisting edgecases will resolve this in nuclei but i think it will be good if we handle such cases in urlutil itself . what do you think @Mzack9999 @RamanaReddy0M
> 
> Not sure but since this issue is in lib itself . non-raw/ base requests might also have this issue
> 
> i think the issue is somewhere in this function https://github.com/projectdiscovery/utils/blob/5efadaf067b4ba4da7149465e52361449e46cce1/url/utils.go#L38-L72 .

I guess this is not just the issue with `MergePath` method. wherever we use `url.parse()` it cause same issue.
I guess we should fix it in `urlutil.Parse` and use this `urlutil.Parse` method in all other lib function like `MergePath`, 
`ParseRelativePath` etc.
what do you think @tarunKoyalwar?

ex: `ParseRelativePath`
https://github.com/projectdiscovery/utils/blob/5efadaf067b4ba4da7149465e52361449e46cce1/url/url.go#L314

**Comment by tarunKoyalwar** [comment]
@RamanaReddy0M yeah , when we parse urls with `url.Parse` it internally normalizes paths similar to how `filepath.Clean()` works . to avoid this we created our version of urlutil . 

+1 for fixing this in urlutil . this is one of the reason why we are replacing `url.Parse` with `urlutil.Parse` everywhere 

In urlutil most of functions have `unsafe bool` parameter and when unsafe is true . we parse it manually instead of using url.Parse()

File: integration_tests/http/raw-unsafe-path.yaml [hunk 1]
```diff
@@ -50,6 +51,10 @@ requests:
       - |+
         GET /xyz/%u2s/%invalid HTTP/1.1
         Host: scanme.sh
+    # test relative path start with // (should not be removed)
+      - |+
+        GET //CFIDE/wizards/common/utils.cfc HTTP/1.1
```
Comment (Mzack9999) [inline integration_tests/http/raw-unsafe-path.yaml:56]: While this represent a nice addition also for `unsafe` requests. I think the integration test is specific for templates having `unsafe` flag not set. We can add a new integration test `raw-path.yaml` (if no equivalent exists already) where progressively track all these edge cases.

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
integration test should cover `unsafe: false` (usually omitted)

File: integration_tests/http/raw-unsafe-path.yaml (hunk 1 above)
Comment (tarunKoyalwar) [reply integration_tests/http/raw-unsafe-path.yaml:56]: yeah , @RamanaReddy0M we also have integration test will above cases with `unsafe: false` at https://github.com/projectdiscovery/nuclei/blob/main/integration_tests/http/http-paths.yaml

both in base and raw format

**Review by Mzack9999 — APPROVED** [review]
lgtm!

**Review by tarunKoyalwar — APPROVED** [review]

**Review by ehsandeep — APPROVED** [review]
